### PR TITLE
lziprecover: init at 1.22

### PR DESCRIPTION
**Pull request by vlaci**
Maintainer is absent for the original package that's why I haven't put myself in here. I can add myself to both of them if needed

###### Motivation for this change

This is a data recovery tool for `lzip` files. which I am depending on for my current job :)

###### Things done

Tried cross-building to multiple platform successfully but didn't test any of the crossbuilt output. CFLAGS and co. are directly lifted over from the `lzip` defivation.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by c0bw3b — COMMENTED** [review]
It builds and runs properly.

One message during build though:

```session
testing --reproduce...
warning: skipping --reproduce test: lzip not found or not the right version.
Try 'make LZIP_NAME=<name_of_lzip_executable> check'.
```

You need to pass `lzip` in `checkInputs` in order to run all possible tests during check phase.

File: pkgs/tools/compression/lziprecover/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "CXX=${stdenv.cc.targetPrefix}c++";
+
+  doCheck = true;
+  enableParallelBuilding = true;
+
+  meta = {
+    homepage = "https://www.nongnu.org/lzip/lziprecover.html";
+    description = "Data recovery tool for lzip compressed files";
+    license = lib.licenses.gpl2Plus;
+    platforms = lib.platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/lziprecover/default.nix:26]: Missing maintainer.

File: pkgs/tools/compression/lziprecover/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    url = "mirror://savannah/lzip/lziprecover/${pname}-${version}.tar.gz";
+    sha256 = "sha256-/ZWKCXX3cpxE87eE5WaJH3NsPcaDdNvSFJ7mkqFtCGI=";
+  };
+
+  configureFlags = [
+    "CPPFLAGS=-DNDEBUG"
+    "CFLAGS=-O3"
+    "CXXFLAGS=-O3"
+  ] ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform)
+    "CXX=${stdenv.cc.targetPrefix}c++";
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/lziprecover/default.nix:17]: Can we apply this uncondtional?

Comment (vlaci) [reply pkgs/tools/compression/lziprecover/default.nix]: I guess we can, I modified lzip too, from which this idiom was copied.

**Comment by SuperSandro2000** [comment]
We can either put this entire PR onto staging or only move the lzip things there because they cause to many rebuilds

**Comment by SuperSandro2000** [comment]
We can either change the entire PR to target staging or only move the lzip changes to staging.

**Comment by vlaci** [comment]
I've retargeted the PR to the `staging` branch

**Pull request merged**